Explore: add URL schema docs (#71724)
grafana/grafana: docs/sources/explore/_index.md

To close the newly created query, click on the Close Split button.

## Share Explore URLs

When using Explore, the URL in the browser address bar is updated as you make changes to the queries. This allows you to share the URLs with others or bookmark them for later use.

### Generating Explore URLs from external tools

> Because Explore URLs have a defined structure, you can build an URL from external tools and open it in Grafana. The URL structure is:

Reviewer: ```suggestion
Because Explore URLs have a defined structure, you can build a URL from external tools and open it in Grafana. The URL structure is:
```

```
http://<grafana_url>/explore?panes=<panes>&schemaVersion=<schema_version>&orgId=<org_id>
```

where:

- `org_id` is the organization ID

> - `schema_version` is the schema version, currently only `1` is supported

Reviewer 2: If we go with mentioning compact URLs below, we should clarify this a little. 

"`schema_version` is the schema version. Currently, we migrate URLs to version `1`"

Author: similarly to https://github.com/grafana/grafana/pull/71724#discussion_r1267924296, i think the objective here is to describe how to generate one for the current version without providing too many details about the whys.

probably however ``currently only `1` is supported`` is a bit confusing as it implies that there will be other schema versions with the same structure. maybe we should rephrase this to something like:

```suggestion
- `schema_version` is the schema version of the URL format, to make Explore parse the URL using the current schema it must be set to `1`
```

maybe? wdyt? (cc. @ifrost @harisrozajac )

Reviewer 3: > - `schema_version` is the schema version of the URL format, to make Explore parse the URL using the current schema it must be set to `1` 

@Elfo404 I think this is a good description. Since it implies that there are other schema versions, would it make sense to specify somewhere what those are?

Reviewer: Maybe?

```suggestion
- `schema_version` is the schema version (should be set to the latest version which is `1`)
```

- `panes` is a url-encoded JSON object of panes, where each key is the pane ID and each value is an object matching the following schema:

```ts
{
  datasource: string; // the datasource UID, or `-- Mixed --` for mixed datasources
  queries: {
    // the queries for this pane
    refId: string; // an alphanumeric identifier for this query, must be unique within the pane, i.e. "A", "B", "C", etc.

>     datasource: {

Reviewer: Is it actually required when a non-mixed data source is used at the root? If not maybe we can mention it, otherwise it might be confusing why it needs to be set up twice when non-mixed is used.

Author: not strictly required as Explore will use the current root datasource if the query doen't have the datasource property, but that was done for legacy reasons. when not defining a datasource in each query explore will update the URL to include on for each, so effectively the URL will match this schema, not one without that property

Reviewer: Then maybe we could add a bit more context for both, e.g.:

```typescript
datasource: string; // Explore's root datasource UID, or `-- Mixed --` for mixed datasources
...
uid: string; // query's datasource UID ie: "AD7864H6422"
```

Author: updated in d05619da00

      uid: string; // the datasource UID ie: "AD7864H6422"
      type: string; // the datasource type-id, i.e: "loki"
    }
    // ... any other datasource-specific query parameters
  }

>   [];

Reviewer: I guess `queries` value escaped from that array!

Author: this is how prettier formats that `[]` after the object if defined as typescript. 
i removed the file type and updated the block in 7998e094af, should be a bit better now

  range: {
    from: string | number; // the start time, in milliseconds since epoch
    to: string | number; // the end time, in milliseconds since epoch
  }
}
```

> **Note**: the `from` and `to` also accepts relative ranges defined in [Time units and relative ranges](https://grafana.com/docs/grafana/latest/dashboards/use-dashboards/?pg=blog&plcmt=body-txt#time-units-and-relative-ranges)

> 

Reviewer 2: Would we want to add a comment here about schema version 0 being for compact URLs? 

"Some older Explore URLs may be in "compact" format, which look like [...]. While we do support these URLs, we migrate the URLs to be in schema version 1. We recommend replacing the source of compact format URLs to match the format defined above."

Author: we discussed this briefly yesterday during standup, the idea for this doc is to only describe the current version as a guide for who wants to build a valid Explore URL. older schemas may better live in types definitions for when we will actually use the schema generator and push those types in `@grafana/schema`, or in older versions of the docs

Reviewer: I think it makes sense to have official documentation only for the latest. We support older versions because of backward compatibility, but they are not officially supported as alternatives.

Because Explore URLs have a defined structure, you can build a URL from external tools and open it in Grafana. The URL structure is:

```
http://<grafana_url>/explore?panes=<panes>&schemaVersion=<schema_version>&orgId=<org_id>
```

where:

- `org_id` is the organization ID
- `schema_version` is the schema version (should be set to the latest version which is `1`)
- `panes` is a url-encoded JSON object of panes, where each key is the pane ID and each value is an object matching the following schema:

```
{
  datasource: string; // the pane's root datasource UID, or `-- Mixed --` for mixed datasources
  queries: {
    refId: string; // an alphanumeric identifier for this query, must be unique within the pane, i.e. "A", "B", "C", etc.
    datasource: {
      uid: string; // the query's datasource UID ie: "AD7864H6422"
      type: string; // the query's datasource type-id, i.e: "loki"
    }
    // ... any other datasource-specific query parameters
  }[]; // array of queries for this pane
  range: {
    from: string | number; // the start time, in milliseconds since epoch
    to: string | number; // the end time, in milliseconds since epoch
  }
}
```

> > **Note**: the `from` and `to` also accepts relative ranges defined in [Time units and relative ranges]({{< relref "../dashboards/use-dashboards/#time-units-and-relative-ranges" >}})

Reviewer 4: I'm updating the note to use the admonition shortcode. We changed the Note format to be an admonition shortcode (see [Shortcodes](https://grafana.com/docs/writers-toolkit/write/shortcodes/#admonition-shortcode) content in the Writers' Toolkit). 
```suggestion
{{% admonition type="note" %}}
The `from` and `to` also accept relative ranges defined in [Time units and relative ranges]({{< relref "../dashboards/use-dashboards/#time-units-and-relative-ranges" >}}).
{{% /admonition %}}
```

Author: thanks!

To close the newly created query, click on the Close Split button.

## Share Explore URLs

> When using Explore, the URL in the browser address bar is updated as you make changes to the queries. This allows you to share the URLs with others or bookmark them for later use.

Reviewer 4: This suggestion moves the phrasing towards active 
```suggestion
When using Explore, the URL in the browser address bar updates as you make changes to the queries. You can share or bookmark this URL. 
```
